libs/blocks/aside/aside.js

```diff
@@ -14,74 +14,77 @@
-  if (!el) return;
-  const text = el.querySelector('h1, h2, h3, h4, h5, h6, a')?.closest('div');
+  const text = foreground.querySelector('h1, h2, h3, h4, h5, h6, a:not(.video)')?.closest('div');
+  text?.classList.add('text');
   const picture = text?.querySelector('picture');
   const iconArea = picture ? (picture.closest('p') || createTag('p', null, picture)) : null;
-  text?.classList.add('text');
-  if (text && !text.querySelector('p')) {
-    const buttons = text.querySelectorAll('em a, strong a');
-    let btnWrap = null;
-    if (buttons[0] && !buttons[0].closest('p')) {
-      btnWrap = document.createElement('p');
-      btnWrap.append(...[...buttons].map((button) => button.parentElement));
-    }
-    const desc = createTag('p', null, text.innerHTML);
-    text.innerHTML = '';
-    text.append(
-      iconArea || '',
-      desc,
-      btnWrap || '',
-    );
-  }
   iconArea?.classList.add('icon-area');
-  decorateButtons(el);
-  const headings = text?.querySelectorAll('h1, h2, h3, h4, h5, h6');
-  const heading = headings?.[headings.length - 1];
-  const isInline = type === INLINE;
-  const isNotification = type === NOTIFICATION;
-  if (heading) {
-    let headingClass = 'heading-xl';
-    if ((isNotification && size === SIZE_M) || isInline) {
-      headingClass = 'heading-s';
+  const foregroundImage = foreground.querySelector(':scope > div:not(.text) img')?.closest('div');
+  const bgImage = el.querySelector(':scope > div:not(.text) img')?.closest('div');
+  const image = foregroundImage ?? bgImage;
+  if (image && !image.classList.contains('text')) {
+    const isSplit = el.classList.contains('split');
+    image.classList.add(`${isSplit ? 'split-' : ''}image`);
+    if (isSplit) {
+      const position = Array.from(image.parentNode.children).indexOf(image);
+      el.classList.add(`split${!position ? '-right' : '-left'}`);
+      foreground.parentElement.appendChild(image);
     }
-    if (isNotification && size === SIZE_L) {
-      headingClass = 'heading-l';
-    }
-    heading?.classList.add(headingClass);
-    const prevClasses = heading?.previousElementSibling?.classList;
-    if (prevClasses?.length === 0) prevClasses.add('detail-m');
+  } else if (!iconArea) {
+    foreground?.classList.add('no-image');
   }
-  const bodyClass = (isNotification && (size === SIZE_XS || size === SIZE_S || size === SIZE_L)) || isInline ? 'body-m' : 'body-s';
-  const bodyCopy = heading?.nextElementSibling.classList.length === 0 ? heading.nextElementSibling : text?.querySelector('p:not([class])');
-  bodyCopy?.classList.add(bodyClass);
-  const body = createTag('div', { class: 'body-area' });
-  bodyCopy?.insertAdjacentElement('beforebegin', body);
-  body.append(bodyCopy);
-  el.querySelector(':scope > div:not(.text) img')?.closest('div').classList.add('image');
+  return foreground;
+}
+
+function decorateVideoContainer(el) {
+  const video = el.querySelector('a[href*=".mp4"]');
+  if (!video) return;
+  video.classList.add('video');
```

Comment:
It looks like the classes are not being added. The reason could be `.mp4` is already decorated by autoblock.

Reply:
Is this being left in? Or is the plan to use @saugatmalla autoblock update for videos? 
Is this for background videos? I don't see any video's in your sample page which makes this all a bit confusing...

cc: @ryanmparrish

Reply:
@rgclayton Only the classes are being added. I believe it is for the positioning of the video.

Reply:
@saugatmalla @rgclayton  I made adjustments to avoid needing this function

Reply:
The decorateVideoContainer function? I still see it as part of your PR.